arch/arm/boot/dts/overlays/rotary-encoder-overlay.dts

```diff
@@ -0,0 +1,27 @@
+// Device tree overlay for GPIO connected rotary encoder.
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2708";
+
```

Comment:
Very close - you just need to add this to the `rotary@0` node:
```
pinctrl-names = "default";
pinctrl-0 = <&rotary0_pins>;
```
This tells the pinctrl subsystem to claim the linked pins on behalf of the device.